ozaria/site/components/interactive/PageInteractive/index.vue

```diff
@@ -1,108 +1,128 @@
+      interactiveComponent () {
+        switch (this.interactive.interactiveType) {
+        case 'draggable-statement-completion':
+          return draggableStatementCompletionComponent
+
+        case 'insert-code':
+          return insertCodeComponent
+
+        case 'draggable-ordering':
+          return draggableOrderingComponent
+
+        default:
+          noty({ text: 'Interactive type is not set for the interactive', type: 'error', timeout: '2000' })
+          throw new Error('Invalid interactive type provided for interactive')
+        }
+      }
+    },
+
+    watch: {
+      interactiveIdOrSlug: async function () {
+        await this.getInteractiveData()
+      }
+    },
+
+    async created () {
+      if (!me.hasInteractiveAccess()) {
+        alert('You must be logged in as an admin to use this page.')
+        return application.router.navigate('/', { trigger: true })
+      }
+
       await this.getInteractiveData()
-    }
-  },
-  async created() {
-    if (!me.hasInteractiveAccess()) {
-      alert('You must be logged in as an admin to use this page.')
-      return application.router.navigate('/', { trigger: true })
-    }
-    await this.getInteractiveData()
-    this.dataLoaded = true
-  },
-  methods: {
-    onCompleted() {
-      this.$emit('completed')
     },
-    async getInteractiveData() {
-      try {
-        this.interactive = await getInteractive(this.interactiveIdOrSlug)
-        this.interactiveType = this.interactive.interactiveType
-        if (!this.interactiveType) {
-          return Promise.reject("Interactive type is not set for the interactive " + this.interactiveIdOrSlug)
-        }
-        if (!me.isSessionless()) { // not saving progress/session for teachers
-          const getSessionOptions = {
-            introLevelId: this.introLevelId,
-            courseInstanceId: this.courseInstanceId
+
+    methods: {
+      onCompleted () {
+        this.$emit('completed')
+      },
+
+      async getInteractiveData () {
+        this.loading = true
+
+        try {
+          this.interactive = await getInteractive(this.interactiveIdOrSlug)
+          this.interactiveType = this.interactive.interactiveType
+          if (!this.interactiveType) {
+            return Promise.reject('Interactive type is not set for the interactive ' + this.interactiveIdOrSlug)
+          }
+          if (!me.isSessionless()) { // not saving progress/session for teachers
+            const getSessionOptions = {
+              introLevelId: this.introLevelId,
+              courseInstanceId: this.courseInstanceId
+            }
+            this.interactiveSession = await getSession(this.interactiveIdOrSlug, getSessionOptions )
           }
-          this.interactiveSession = await getSession(this.interactiveIdOrSlug, getSessionOptions )
+        } catch (err) {
+          console.error('Error:', err)
+          return noty({ text: 'Error occured in creating interactives data.', type: 'error', timeout: '2000' })
```

Comment:
I would be interested in your opinions of how we can use errors. Is noty what you recommend?
Should we tell the user to refresh the browser?